spec/lib/reporting_spec.rb

```diff
@@ -2,102 +2,114 @@
 require 'reporting'
 
 RSpec.describe Reporting do
-  subject(:reporting) do
-    Class.new { extend Reporting }
-  end
-
-  let(:user1) do
-    create(:user, :registered, email: 'test1@example.com')
-  end
-
-  let(:user2) do
-    create(:user, :registered, email: 'test2@example.com')
-  end
-
-  let(:user1_note1) do
-    user1.notes.create(body: 'test note body')
-  end
+  describe '.users' do
+    subject(:reporting) { Class.new { extend Reporting } }
 
-  let(:user1_note2) do
-    user1.notes.create(body: 'test note body 2')
-  end
+    let(:user1) { create(:user, :registered, email: 'test1@example.com') }
+    let(:user2) { create(:user, :registered, email: 'test2@example.com') }
 
-  let(:user2_note) do
-    user2.notes.create(body: 'test note body')
-  end
+    describe '.users' do
+      let(:with_notes) { reporting.users[:with_notes] }
+      let(:with_notes_percentage) { reporting.users[:with_notes_percentage] }
+      let(:without_notes) { reporting.users[:without_notes] }
+      let(:without_notes_percentage) { reporting.users[:without_notes_percentage] }
 
-  let(:blank_note) do
-    user1.notes.create(body: '')
-  end
+      let(:note_1) { user1.notes.create(body: 'test note body') }
+      let(:note_2) { user1.notes.create(body: 'test note body 2') }
+      let(:note_3) { user2.notes.create(body: 'test note body') }
 
-  let(:whitespace_note) do
-    user1.notes.create(body: " \n \n \n")
-  end
+      context 'when 0 users or notes exist' do
+        it 'with_notes is 0 and percentage is NaN' do
+          expect(with_notes).to be_zero
+          expect(with_notes_percentage).to be_nan
+        end
+      end
 
-  describe '.users' do
-    context 'when 0 users or notes exist' do
-      it 'notes count is 0 and percentage is NaN' do
-        expect(reporting.users[:with_notes]).to eq(0)
-        expect(reporting.users[:with_notes_percentage].nan?).to eq(true)
+      context 'when there are users but no notes' do
+        it 'with_notes count and percentage are 0 and without notes is 100%' do
+          expect(user1.notes.count).to be_zero
+          expect(with_notes).to eq(0)
+          expect(with_notes_percentage).to be_zero
+          expect(without_notes).to eq(1)
+          expect(without_notes_percentage).to eq(100.0)
+        end
       end
-    end
 
-    context 'when there are users but no notes' do
-      it 'notes count is 0 and percentage is 0' do
-        expect(user1.email).to eq('test1@example.com')
-        expect(reporting.users[:with_notes]).to eq(0)
-        expect(reporting.users[:with_notes_percentage]).to eq(0)
+      context 'when there is a saved note but the body is empty' do
+        let(:blank_note) { user1.notes.create(body: '') }
+        let(:whitespace_note) { user1.notes.create(body: " \n \n \n") }
+
+        it 'the note is not counted' do
+          expect(blank_note.filled?).to eq(false)
```

Comment:
We still have assertions that don't need to be made in some of these specs, like checking what the body contains, we should be confident as we have just used a factory to make it. However I agree we do need to test the `#filled?` method. It is a `Note` model instance predicate though so that's the spec we should do it in. As such we can also make use of some syntactic sugar and make it read:

```ruby
expect(note).to be_filled?
```